pkg/controller/jobframework/reconciler.go

```diff
@@ -544,28 +556,25 @@ func FindMatchingWorkloads(ctx context.Context, c client.Client, job GenericJob)
 	return match, toDelete, nil
 }
 
-func (r *JobReconciler) getPrebuiltWorkload(ctx context.Context, job GenericJob, object client.Object, name string) (*kueue.Workload, error) {
-	wl := &kueue.Workload{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: name, Namespace: object.GetNamespace()}, wl)
-	if err != nil {
-		return nil, client.IgnoreNotFound(err)
-	}
-
+func (r *JobReconciler) ensurePrebuiltWorkloadOwnership(ctx context.Context, wl *kueue.Workload, object client.Object) (bool, error) {
 	if !metav1.IsControlledBy(wl, object) {
 		if err := ctrl.SetControllerReference(object, wl, r.client.Scheme()); err != nil {
 			// don't return an error here, since a retry cannot give a different result
```

Comment:
I don't think we should be silencing the error. There might be different ones, and they may be important to investigate. We should log them at least. I think, ideally, we would mark the workload finished, as we do here for other non-retriable errors: https://github.com/kubernetes-sigs/kueue/blob/4cff47bd12e0915680bb9a0f3d0e10d9b284912a/pkg/controller/jobframework/reconciler.go#L354-L361

Reply:
logged